code/_onclick/item_attack.dm

```diff
@@ -13,7 +13,19 @@
 	if(tool_behaviour && target.tool_act(user, src, tool_behaviour))
 		return TRUE
 
-	if(pre_attack(target, user, params))
+	var/pre_attack_result
+	if (is_right_clicking)
+		switch (pre_attack_alt(target, user, params))
+			if (ALT_ATTACK_CALL_NORMAL)
+				pre_attack_result = pre_attack(src, user, params)
+			if (ALT_ATTACK_CANCEL_ATTACK_CHAIN)
+				return TRUE
+			if (null)
```

Comment:
```suggestion
			else
```

Reply:
This on its own would crash for valid inputs. I'll make it so it doesn't by adding an extra check for `if (ALT_ATTACK_CONTINUE_CHAIN)`.